FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.JobManagement/AzureStorageClientFactory.cs

```diff
@@ -21,31 +21,34 @@ public class AzureStorageClientFactory : IAzureStorageClientFactory
         private readonly string _tableName;
         private readonly string _queueUrl;
         private readonly string _queueName;
+        private readonly string _internalConnectionString;
 
         public AzureStorageClientFactory(
-            IOptions<JobConfiguration> config,
+            IOptions<JobConfiguration> jobConfiguration,
+            IOptions<StorageConfiguration> storageConfiguration,
             ITokenCredentialProvider credentialProvider)
         {
-            EnsureArg.IsNotNull(config, nameof(config));
-            EnsureArg.IsNotNullOrWhiteSpace(config.Value.TableUrl, nameof(config.Value.TableUrl));
-            EnsureArg.IsNotNullOrWhiteSpace(config.Value.QueueUrl, nameof(config.Value.QueueUrl));
-            EnsureArg.IsNotNullOrWhiteSpace(config.Value.AgentName, nameof(config.Value.AgentName));
+            EnsureArg.IsNotNull(jobConfiguration, nameof(jobConfiguration));
+            EnsureArg.IsNotNull(storageConfiguration, nameof(storageConfiguration));
+            _tableUrl = EnsureArg.IsNotNullOrWhiteSpace(jobConfiguration.Value.TableUrl, nameof(jobConfiguration.Value.TableUrl));
+            EnsureArg.IsNotNullOrWhiteSpace(jobConfiguration.Value.QueueUrl, nameof(jobConfiguration.Value.QueueUrl));
+            EnsureArg.IsNotNullOrWhiteSpace(jobConfiguration.Value.AgentName, nameof(jobConfiguration.Value.AgentName));
 
-            _tableUrl = config.Value.TableUrl;
-            _tableName = AzureStorageKeyProvider.JobInfoTableName(config.Value.AgentName);
+            _tableName = AzureStorageKeyProvider.JobInfoTableName(jobConfiguration.Value.AgentName);
+            _internalConnectionString = storageConfiguration.Value.InternalStorageConnectionString;
 
-            if (config.Value.QueueUrl != StorageEmulatorConnectionString)
+            if (jobConfiguration.Value.QueueUrl != StorageEmulatorConnectionString)
             {
                 // If the baseUri has relative parts (like /api), then the relative part must be terminated with a slash (like /api/).
                 // Otherwise the relative part will be omitted when creating new Uri with queue name. See https://docs.microsoft.com/en-us/dotnet/api/system.uri.-ctor?view=net-6.0
-            _queueUrl = config.Value.QueueUrl.EndsWith("/") ? config.Value.QueueUrl : $"{config.Value.QueueUrl}/";
+            _queueUrl = jobConfiguration.Value.QueueUrl.EndsWith("/") ? jobConfiguration.Value.QueueUrl : $"{jobConfiguration.Value.QueueUrl}/";
```

Comment:
not sure why we use the table url directly but add a slash to queue url? Is there any difference between these two urls?

Reply:
Queue client needs concat url and name when create queue client, whereas table client does not.